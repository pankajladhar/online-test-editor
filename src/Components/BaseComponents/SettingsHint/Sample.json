[
    {
      "question": "Question1",
      "answers": ["Answer11", "Answer12"],
      "correctAnswer": "Answer11"
    },
    {
      "question": "Question2",
      "answers": ["Answer21", "Answer22",  "Answer23"],
      "correctAnswer": "Answer21"
    }
]