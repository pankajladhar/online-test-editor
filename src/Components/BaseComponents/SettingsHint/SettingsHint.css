.SettingsHint {
  padding: 20px; }
  .SettingsHint .SettingsHint__List {
    padding-bottom: 20px; }
    .SettingsHint .SettingsHint__List span {
      display: inline-block; }
  .SettingsHint .SettingsHint__List__Count {
    background: #fff;
    border-radius: 50%;
    width: 25px;
    height: 25px;
    text-align: center;
    padding-top: 2px; }
  .SettingsHint .SettingsHint__List__StepName {
    font-size: 25px;
    font-weight: bold;
    padding-left: 10px;
    padding-bottom: 5px; }
  .SettingsHint .SettingsHint__List__Optional {
    font-size: 14px;
    padding-left: 10px;
    color: #fff; }
  .SettingsHint .SettingsHint__List__Description,
  .SettingsHint .SettingsHint__List__SubDescription {
    font-size: 16px;
    padding-left: 36px; }
  .SettingsHint .SettingsHint__List__SubDescription {
    font-size: 16px;
    display: block;
    color: #fff; }
    .SettingsHint .SettingsHint__List__SubDescription a {
      color: #1b2133;
      text-decoration: underline; }
      .SettingsHint .SettingsHint__List__SubDescription a:hover {
        text-decoration: none; }
