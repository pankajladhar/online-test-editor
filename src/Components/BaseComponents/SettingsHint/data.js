const data = [
    {
        "stepName": "Quiz title",
        "isMandatory": false,
        "description": "This is will be shown in header. It is optional, can be empty. In case of empty it will show  Quizbox"
    },
    {
        "stepName": "Question url",
        "isMandatory": true,
        "description": "This will be used to fetch questions. This must be a public URL.",
        "subDescription" : [
            "github can be used to create a public URL.(/configure/github) [Click Here]",
            "<a href='./Sample.json' download>Download Sample</a>"
        ]
    },
    {
        "stepName": "No of questions",
        "isMandatory": true,
        "description": "This is will be used to get total questions. ex: If you have 100 questions and you want to get only 10 questions at a time give 10"
    },
    {
        "stepName": "Points of per question",
        "isMandatory": true,
        "description": "This is will be used to calculate score."
    },
    {
        "stepName": "Author Name",
        "isMandatory": false,
        "description": "This is will be shown in header as well as in sample tab. It is optional, can be empty. In case of empty it will show anything"
    },
]

export default data;