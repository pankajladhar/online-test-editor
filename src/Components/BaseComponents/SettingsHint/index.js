import React, { Component } from 'react';
import PropTypes from 'prop-types';
import data from './data';
import './SettingsHint.css';


class SettingsHint extends Component {
    render() {
        return (
            <div className="SettingsHint">
                {
                    data.map((item, index) => {
                        return (
                            <div className="SettingsHint__List" key={index}>
                                <span className="SettingsHint__List__Count">&#10004;</span>
                                <span className="SettingsHint__List__StepName">{item.stepName}</span>
                                {item.isMandatory && <span className="SettingsHint__List__Optional">optional</span>}
                                <span className="SettingsHint__List__Description">{item.description}</span>
                                {item.subDescription && <span className="SettingsHint__List__SubDescription" dangerouslySetInnerHTML={{__html: item.subDescription}}/>}
                            </div>
                        )
                    })
                }
            </div>
        );
    }
}

SettingsHint.propTypes = {

};

export default SettingsHint;