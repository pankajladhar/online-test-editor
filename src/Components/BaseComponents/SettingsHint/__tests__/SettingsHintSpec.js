import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import SettingsHint from './../index';

describe('SettingsHint', ()=>{
    it('renders correctly', ()=>{
        let tree = renderer.create(
            <SettingsHint/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});