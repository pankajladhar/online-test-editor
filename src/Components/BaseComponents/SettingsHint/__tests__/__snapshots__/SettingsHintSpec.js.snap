// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsHint renders correctly 1`] = `
<div
  className="SettingsHint"
>
  <div
    className="SettingsHint__List"
  >
    <span
      className="SettingsHint__List__Count"
    >
      ✔
    </span>
    <span
      className="SettingsHint__List__StepName"
    >
      Quiz title
    </span>
    <span
      className="SettingsHint__List__Description"
    >
      This is will be shown in header. It is optional, can be empty. In case of empty it will show  Quizbox
    </span>
  </div>
  <div
    className="SettingsHint__List"
  >
    <span
      className="SettingsHint__List__Count"
    >
      ✔
    </span>
    <span
      className="SettingsHint__List__StepName"
    >
      Question url
    </span>
    <span
      className="SettingsHint__List__Optional"
    >
      optional
    </span>
    <span
      className="SettingsHint__List__Description"
    >
      This will be used to fetch questions. This must be a public URL.
    </span>
    <span
      className="SettingsHint__List__SubDescription"
      dangerouslySetInnerHTML={
        Object {
          "__html": "github can be used to create a public URL. <a href=''>Click Here</a>",
        }
      }
    />
  </div>
  <div
    className="SettingsHint__List"
  >
    <span
      className="SettingsHint__List__Count"
    >
      ✔
    </span>
    <span
      className="SettingsHint__List__StepName"
    >
      No of questions
    </span>
    <span
      className="SettingsHint__List__Optional"
    >
      optional
    </span>
    <span
      className="SettingsHint__List__Description"
    >
      This is will be used to get total questions. ex: If you have 100 questions and you want to get only 10 questions at a time give 10
    </span>
  </div>
  <div
    className="SettingsHint__List"
  >
    <span
      className="SettingsHint__List__Count"
    >
      ✔
    </span>
    <span
      className="SettingsHint__List__StepName"
    >
      Points of per question
    </span>
    <span
      className="SettingsHint__List__Optional"
    >
      optional
    </span>
    <span
      className="SettingsHint__List__Description"
    >
      This is will be used to calculate score.
    </span>
  </div>
  <div
    className="SettingsHint__List"
  >
    <span
      className="SettingsHint__List__Count"
    >
      ✔
    </span>
    <span
      className="SettingsHint__List__StepName"
    >
      Author Name
    </span>
    <span
      className="SettingsHint__List__Description"
    >
      This is will be shown in header as well as in sample tab. It is optional, can be empty. In case of empty it will show anything
    </span>
  </div>
</div>
`;
